### PR TITLE
Feature/add song from youtube search

### PR DESCRIPTION
**Pull request by goody153**
## Ticket 
  - [Add song from youtube search](https://trello.com/c/PcE1rLHm)

## Summary
  - added button and dropdown on playlist.html
  - added new class on view and urls
  - added restriction for youtube searches to appear if the search keyword is empty
  - allowed songForm validations when adding through youtubesearch

### CODE REVIEW

File: playlist/views.py [hunk 1]
```diff
@@ -179,3 +181,32 @@ def post(self, *args, **kwargs):
...
+        form = SongForm(
+            self.request.POST,
+            user=self.request.user,
+            playlist=playlist,
+        )
+        if form.is_valid():
+            form.save()
+            return redirect('playlist', self.request.POST.get('playlist'))
+        # return validation error on playlist
+        songs = Song.objects.filter(playlist=playlist, archive=False)
```
Comment (mcandidier) [inline playlist/views.py]: i think you don't need to get songs from the playlist. If so, please explain why you add songs.

File: playlist/urls.py [hunk 2]
```diff
@@ -17,4 +18,5 @@
     url(r'^playlist/(?P<playlist_id>\d+)/(?P<song_id>\d+)/delete/$', SongDelete.as_view(), name='song_delete'),
     url(r'^playlist/search/$', SearchSongYoutube.as_view(), name='search_youtube'),
     url(r'^playlist/playlistsearch/$', SearchedPlaylist.as_view(), name='search_playlist'),
+    url(r'^playlist/search/addtoyoutube/$', AddToPlaylistFromYoutube.as_view(), name='add_youtube_search'),
```
Comment (mcandidier) [inline playlist/urls.py]: make it short, playlist/search/add

File: playlist/views.py (hunk 1 above)
Comment (goody153) [reply playlist/views.py]: Updated.

File: playlist/urls.py (hunk 2 above)
Comment (goody153) [reply playlist/urls.py]: Updated.

File: playlist/views.py [hunk 3]
```diff
@@ -179,3 +181,24 @@ def post(self, *args, **kwargs):
...
+            id=self.request.POST.get('playlist')
+        )
+        form = SongForm(
+            self.request.POST,
+            user=self.request.user,
+            playlist=playlist,
+        )
+        if form.is_valid():
+            form.save()
+            return JsonResponse({'result':'Successfully added song to playlist.'}, safe=False)
```
Comment (mcandidier) [inline playlist/views.py]: move success message directly to template.

Comment (goody153) [reply playlist/views.py]: Updated.

**Pull request merged**